addons/hr/models/res_users.py

```diff
@@ -185,20 +185,15 @@ def write(self, vals):
             raise AccessError(_("You are only allowed to update your preferences. Please contact a HR officer to update other information."))
 
         result = super(User, self).write(vals)
-
         employee_values = {}
         for fname in [f for f in ['name', 'email', 'image_1920', 'tz'] if f in vals]:
             employee_values[fname] = vals[fname]
         if employee_values:
             if 'email' in employee_values:
                 employee_values['work_email'] = employee_values.pop('email')
             if 'image_1920' in vals:
-                without_image = self.env['hr.employee'].sudo().search([('user_id', 'in', self.ids), ('image_1920', '=', False)])
-                with_image = self.env['hr.employee'].sudo().search([('user_id', 'in', self.ids), ('image_1920', '!=', False)])
-                without_image.write(employee_values)
-                if not can_edit_self:
-                    employee_values.pop('image_1920')
-                with_image.write(employee_values)
+                employee_values['image_placeholder_1920'] = employee_values.pop(fname)
```

Comment:
Wouldn't this fail to detect writing through the inverse on `image_1920`?

Also this code looks wrong, `fname` is defined in the loop above, but this condition is not part of it.

Reply:
Yes, indeed the code was wrong. Nice catch. Regarding detecting writing on `image_1920`, we want the value to be written in `image_placeholder_1920`

Reply:
If you write an inverse for `image_1920` that means you are expecting people to use it. If you really want to force writing only through `image_placeholder_1920` you should not add any inverse at all.